test/e2e/loro-os-2-0-single.js

```diff
@@ -0,0 +1,483 @@
+var assert = require('assert')
+var chai = require('chai')
+var chaiHttp = require('chai-http')
+var app = require('../../restApp.js')
+var should = chai.should()
+var setup = require('./setup.js')
+var appLogger = require('../../rest/lib/appLogger.js')
+var request = require('request')
+
+chai.use(chaiHttp)
+var server = chai.request(app).keepOpen()
+
+describe('E2E Test for Single LoraOS 2.0', function () {
+  var adminToken
+  var userId
+  var userToken
+  var loraProtocolId
+  var networkId
+  var applicationId
+
+  before((done) => {
+    setup.start()
+      .then(() => {
+        done()
+      })
+      .catch((err) => {
+        done(err)
+      })
+  })
```

Comment:
You can write this as `setup.start().then(done, done)`

Reply:
I thought then(done, done) only worked if you declared async(done).  I have used that in places.

Reply:
I'm not familiar with mocha, but I'm guessing you could probably return a promise from the before function instead of using done.  Then it would be `before(() => setup.start())`.  As for the `then` function, the first argument is the success handler and the second one is the error handler.

Reply:
What I have seen in the past is 
```
 it('Verify the Test Application Exists on LPWan', function (done) ... .then(done, done) ->Error
```
```
 it('Verify the Test Application Exists on LPWan', (done) => {}).then(done, done) -> Error
```
```
 it('Verify the Test Application Exists on LPWan', async(done) => {}).then(done, done) ->Okay
```

Reply:
I dont' know why

Reply:
I will try it on the Loriot test